CryptoAnalysis-Android/src/main/java/de/fraunhofer/iem/crypto/CogniCryptAndroidAnalysis.java

```diff
@@ -29,56 +25,154 @@
 import soot.jimple.infoflow.android.config.SootConfigForAndroid;
 import soot.options.Options;
 
+import java.io.File;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 public class CogniCryptAndroidAnalysis {
-	public static void main(String... args) {
-		CogniCryptAndroidAnalysis analysis = new CogniCryptAndroidAnalysis(args[0], args[1], args[2],Lists.<String>newArrayList());
-		analysis.run();
-	}
 
 	private static final Logger logger = LoggerFactory.getLogger(CogniCryptAndroidAnalysis.class);
-	private final String apkFile;
-	private final String platformsDirectory;
-	private final String rulesDirectory;
+	private File apkFile;
```

Comment:
Is there a reason you removed the final declarations?

Reply:
yes, because i'm using a setter thus a filed cannot be final